crates/bevy_reflect/derive/src/from_reflect.rs

```diff
@@ -268,16 +287,50 @@ fn get_active_fields(
                     }
                 };
 
+                let conversions: Vec<_> = field
+                    .attrs
+                    .conversions
+                    .iter()
+                    .map(|conversion| {
+                        let path = &conversion.path;
+
+                        let get_value = into_remote(quote! {
+                            <#path as #bevy_reflect_path::FromReflect>::from_reflect(field)
+                        });
+
+                        match &conversion.func {
+                            Some(func) => quote! {
+                                if let #FQOption::Some(field) = #get_value {
+                                    Some((#func)(field).into())
+                                }
+                            },
+                            None => quote! {
+                                if let #FQOption::Some(field) = #get_value {
+                                    Some(field.into())
+                                }
+                            },
+                        }
+                    })
+                    .collect();
+
                 let value = match &field.attrs.default {
                     DefaultBehavior::Func(path) => {
                         let value = into_remote(quote! {
                             <#ty as #bevy_reflect_path::FromReflect>::from_reflect(field)
                         });
+
+                        let some_branch = if conversions.is_empty() {
+                            quote! { #value }
+                        } else {
+                            quote! { #(#conversions)else* else { #value } }
```

Comment:
Nit: Does `#(#conversions else)* { #value }` work? Because not only would that remove the `else` duplication, but might allow you to do away with `some_branch` altogether.

Reply:
Fixed.